sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs

```diff
@@ -731,6 +731,269 @@ public async Task DownloadEmptyBlobTest()
+                await outputStream.ReadAsync(outputBytes, 0, size);
+
+                // Assert
+                Assert.AreEqual(data.Length, outputStream.Length);
+                TestHelper.AssertSequenceEqual(data, outputBytes);
+            }
+        }
+
+        [Test]
+        public async Task OpenReadAsync_AccessConditionsFail()
+        {
+            // Arrange
+            int size = Constants.KB;
+            var garbageLeaseId = GetGarbageLeaseId();
+            foreach (AccessConditionParameters parameters in GetAccessConditionsFail_Data(garbageLeaseId))
+            {
+                await using DisposingContainer test = await GetTestContainerAsync();
+                var data = GetRandomBuffer(size);
+                BlockBlobClient blob = InstrumentClient(test.Container.GetBlockBlobClient(GetNewBlobName()));
+                using Stream stream = new MemoryStream(data);
+                await blob.UploadAsync(stream);
+
+                parameters.NoneMatch = await SetupBlobMatchCondition(blob, parameters.NoneMatch);
+                BlobRequestConditions accessConditions = BuildAccessConditions(parameters);
+
+                // Act
+                Stream outputStream = await blob.OpenReadAsync(
+                    bufferSize: size / 4,
+                    conditions: accessConditions).ConfigureAwait(false);
+                byte[] outputBytes = new byte[size];
+
+                await TestHelper.CatchAsync<Exception>(
+                    async () =>
+                    {
+                        var _ = await outputStream.ReadAsync(outputBytes, 0, size);
+                    });
+            }
+        }
+
+        [Test]
+        public async Task OpenReadAsync_StrangeOffsetsTest()
+        {
+            // Arrange
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            int size = Constants.KB;
+            byte[] exectedData = GetRandomBuffer(size);
+            BlobClient blobClient = InstrumentClient(test.Container.GetBlobClient(GetNewBlobName()));
+            using Stream stream = new MemoryStream(exectedData);
+            await blobClient.UploadAsync(stream);
+
+            Stream outputStream = await blobClient.OpenReadAsync(position: 0, bufferSize: 157);
+            byte[] actualData = new byte[size];
+            int offset = 0;
+
+            // Act
+            int count = 0;
+            while (offset + count < size)
+            {
+                for (count = 6; count < 37; count += 6)
+                {
+                    await outputStream.ReadAsync(actualData, offset, count);
+                    offset += count;
+                }
+            }
+            await outputStream.ReadAsync(actualData, offset, size - offset);
+
+            // Assert
+            TestHelper.AssertSequenceEqual(exectedData, actualData);
+        }
+
+        [Test]
+        [Ignore("Don't want to record 1 GB of data.")]
+        public async Task OpenReadAsync_LargeData()
```

Comment:
This is the perf test.  Here, we upload a 1GB blob, and then download it via `OpenRead()`.  The `Upload()` section of the test raises our memory usage to ~2GB.  `OpenRead()` appears to use a constant, small amount of memory proportional to the size of the array passed into the `Read()` calls.